src/items.rs

```diff
@@ -113,24 +113,59 @@ impl Rewrite for ast::Local {
         result.push_str(&infix);
 
         if let Some((init, _els)) = self.kind.init_else_opt() {
-            // 1 = trailing semicolon;
-            let nested_shape = shape.sub_width(1)?;
+            let base_span = if let Some(ref ty) = self.ty {
+                mk_sp(ty.span.hi(), self.span.hi())
+            } else {
+                mk_sp(self.pat.span.hi(), self.span.hi())
+            };
```

Comment:
Took another look at this and I realized that the `base_span` is just used to find the comments in `rewrite_initializer_expr`. Instead of using the entire span of the `ast::Local` It probably makes more sense just to use the span between the `pat` or `ty` and the `init` like:

```suggestion
            let base_span = if let Some(ref ty) = self.ty {
                ty.span.between(init.span)
            } else {
                self.pat.span.between(init.span)
            };
```

Possibly renaming `base_span` to better indicate what's its used for.